astropy/io/fits/util.py

```diff
@@ -585,7 +585,7 @@ def _array_from_file(infile, dtype, count, sep):
         # their underlying file object, instead of the decompressed bytes
         read_size = np.dtype(dtype).itemsize * count
         s = infile.read(read_size)
-        return np.fromstring(s, dtype=dtype, count=count, sep=sep)
```

Comment:
This may be wrong: is `sep` ever given as a real string? Certainly, `fromfile` can handle both an empty string (binary) and a given separation string (text). Unless we are sure `sep` is always '', we probably are stuck with
```
if sep == '':
    return np.frombuffer(...)
else:
    return np.fromstring(..., sep=sep)
```

Reply:
OK, I've added this back (GH just doesn't recognize it).

Reply:
Should be OK now, thanks!

Reply:
`sep` is always used with an empty string so I think it's safe to simplify this, see #6813